examples/04_glm_first_level/plot_two_session_model.py

```diff
@@ -0,0 +1,257 @@
+    threshold=3.0,
+    cut_coords=cut_coords,
+    title=f'{contrast_id}, first session',
+)
+
+###############################################################################
+# Compute the statistics for the second session.
+fmri_glm_ses2 = fmri_glm.fit(fmri_imgs[1], design_matrices=design_matrices[1])
+
+contrast_val = pad_vector([-1, -1, 1, 1], design_matrices[1].shape[1])
+
+summary_statistics_ses2 = fmri_glm_ses2.compute_contrast(
+    contrast_val,
+    output_type='all',
+)
+plotting.plot_stat_map(
+    summary_statistics_ses2['z_score'],
+    bg_img=mean_img_,
+    threshold=3.0,
+    cut_coords=cut_coords,
+    title=f'{contrast_id}, second session',
+)
+
+###############################################################################
+# Compute the fixed effects statistics using both sessions' statistical maps.
+#
+# We can use :func:`~nilearn.glm.contrasts.compute_fixed_effects` to compute
+# the fixed effects statistics using the outputs from the session-specific
+# FirstLevelModel results.
+from nilearn.glm.contrasts import compute_fixed_effects
+
+contrast_imgs = [
+    summary_statistics_ses1['effect_size'],
+    summary_statistics_ses2['effect_size'],
+]
+variance_imgs = [
+    summary_statistics_ses1['effect_variance'],
+    summary_statistics_ses2['effect_variance'],
+]
+
+fixed_fx_contrast, fixed_fx_variance, fixed_fx_stat = compute_fixed_effects(
+    contrast_imgs,
+    variance_imgs,
+    data['mask'],
+)
+plotting.plot_stat_map(
+    fixed_fx_stat,
+    bg_img=mean_img_,
+    threshold=3.0,
+    cut_coords=cut_coords,
+    title=f'{contrast_id}, fixed effects',
+)
+
+###############################################################################
+# Not unexpectedly, the fixed effects version displays higher peaks than the
+# input sessions. Computing fixed effects enhances the signal-to-noise ratio of
+# the resulting brain maps.
+
+###############################################################################
+# Compute the fixed effects statistics using both sessions' preprocessed data.
+#
+# A more straightforward alternative to fitting session-specific GLMs, then
+# combining the results with
+# :func:`~nilearn.glm.contrasts.compute_fixed_effects`, is to simply fit the
+# GLM to both sessions at once.
+#
+# Since we can assume that the design matrices of both sessions have the same
+# columns, in the same order, we can again re-use the first session's contrast
+# vector.
+fmri_glm_multises = fmri_glm.fit(fmri_imgs, design_matrices=design_matrices)
+
+# We can just define the contrast array for one session and assume that the
+# design matrix is the same for the other.
+# However, if we want to be safe, we should define each contrast separately,
+# and provide it as a list.
+contrast_val = [
+    pad_vector([-1, -1, 1, 1], design_matrices[0].shape[1]),  # session 1
+    pad_vector([-1, -1, 1, 1], design_matrices[1].shape[1]),  # session 2
+]
+
+z_map = fmri_glm_multises.compute_contrast(
+    contrast_val,
```

Comment:
note that we could specify contrasts symbolically, which would make this part of the example much clearer.

Reply:
Agreed. The design matrices don't have column names, though. Are the condition names enumerated somewhere?

Reply:
You're right. This smells like old-, non-bids-compliant datasets. We could improve that though by providing names when we create the dataframes. I'll open an issue.

Reply:
Thanks! I was planning to infer the appropriate column names from the contrast definitions, but I wasn't sure how to interpret the abbreviations.

Reply:
Given that the symbolic contrast issue has been punted to #3208, is it alright if we stick with the array-based contrast definitions for now?

Reply:
Yes.